include/acl.h

```diff
@@ -478,6 +479,13 @@ typedef class acl_device_program_info_t *acl_device_program_info;
  */
 #define ACL_MEM_CAPABILITY_P2P (1 << 3)
 
+// Definition of device global.
+typedef struct acl_device_global_mem_def_t {
+  std::string name;
+  unsigned int address;
+  unsigned int size;
```

Comment:
Could you confirm this is indeed the desired integer type? `unsigned int` basically says that you want an unsigned integer that is *at least* 32 bits (on the architectures supported by the runtime) but might be larger. `uint32_t` would be more appropriate for *exactly* 32 bits.

Reply:
size could technically be uint32_t, I applied the change.

Reply:
Why is `size` 32 bit and not 64 bit? Does this also apply to `address`?

Reply:
On the frontend, they defined the size as 32 bit. https://github.com/intel/llvm/blob/88123c1c42e71aa3667d24558ed9c9baca39467a/llvm/tools/sycl-post-link/DeviceGlobals.cpp#L61

Reply:
I think address is going to be converted from some pointer value, Its safer to keep it unsigned int

Reply:
Thanks for confirming `size`. As for `address`, if it stores a pointer value you might want to use `uintptr_t`. But I would recommend you do try to find the corresponding type in the compiler and use that here.

Reply:
I checked the IR generated for device global, if I am interpreting it correctly, the type should be uint32 for address as well: https://github.com/intel/llvm/blob/88123c1c42e71aa3667d24558ed9c9baca39467a/llvm/test/tools/sycl-post-link/device-globals/test_no_property_set_header_for_an_empty_set.ll#L11